### PR TITLE
fix(DB/spell_dbc): The Green Tower equip spell trigger

### LINKED ISSUE #5649: The Green Tower Shield - Proc Does Not Work
#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [x] enUS
- [ ] other (specify)

##### FACTION

- [x] Alliance
- [ ] Horde

##### CONTENT PHASE:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [x] 30-39
- [ ] 40-49
- [ ] 50-59

##### CURRENT BEHAVIOUR:

Original report: https://github.com/chromiecraft/chromiecraft/issues/531

The Equip on the lvl 36 item The Green Tower does not proc. https://wowgaming.altervista.org/aowow/?item=1204

Tester's notes - The proc is only a 1% rate anyway, but tried this with an 80 warrior wearing the shield vs a fast-hitting lvl 20 druid cat, and let the cat melee the warr for about 10 mins, delivering about 600 melee attacks. No procs were seen before the warr died, so it does seem pretty likely the proc is not working.

WoWScrnShot_050521_231513

##### EXPECTED BLIZZLIKE BEHAVIOUR:

" When struck in combat has a 1% chance of raising a thorny shield that inflicts 3 Nature damage to attackers when hit and increases Nature resistance by 50 for 30 sec." 

The spell it should cast is https://wowgaming.altervista.org/aowow/?spell=18097

##### STEPS TO REPRODUCE THE PROBLEM:

1. Have a high level character with the shield duel a low level character (so they don't do much damage)
2. Gather a few hundred melee strikes.

##### EXTRA NOTES:

https://wowgaming.altervista.org/aowow/?item=1204

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/91081f4ad8fed98d7bdc9c0ae13d05c86032df9f

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/98292739-the-green-tower-shield-proc-does-not-work?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Maxproz**
**Description:**
Issue was reported saying the green tower(30-39) shield spell proc was not going off at all.
Investigated the spell inside spell editor and saw inconsistent values on the trigger chances.
I changed the values by making comparisons to skullflame shield and the issue appears to be fixed.

## Changes Proposed:
-  Database spell overrides for the TheGreenTower aura ability and its related trigger.

## Issues Addressed:

- Closes #5649

## Tests Performed:
Issue built without errors, I personally did the repro steps below and got 2 procs back to back after some time.

## How to Test the Changes:

1. Spawn a Shaman in durotar
2. Make him level 60
3. Give yourself item 1204
4. Equip item 1204
5. Run up to a boar
6. punch it in the face
7. let it auto attack you 
8. observe you will get equip proc eventually 

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Si1ker — APPROVED** [review]

**Comment by Maxproz** [comment]
FrancescoBorzi 
16 hours ago Member
hello! thanks for your contribution, please adjust your queries according to: https://www.azerothcore.org/wiki/spell_dbc

also the 2 statements INSERT IGNORE can be combined into one

**Comment by FrancescoBorzi** [comment]
@Maxproz I hope is clear what I meant

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1620335015933017000.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
...
+		(18097,0,0,0,192,268435456,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,40,1,0,0,0,0,9,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,-1,0,0,6,0,0,1,0,0,0,0,0,-1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,42,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,17154,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,"The Green Tower","","","","","","","","","","","","","","","",16712190,"","","","","","","","","","","","","","","","",16712188,"When struck in combat has a 1% chance of raising a thorny shield that inflicts $17154s1 Nature damage to attackers when hit and increases Nature resistance by $17154s2 for $17154d.","","","","","","","","","","","","","","","",16712190,"","","","","","","","","","","","","","","","",16712188,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,0,0,0,0,0,0,32,0,0,0,0,0,0,0,0);
+UPDATE spell_dbc SET ProcTypeMask = '0', ProcChance = '101' where ID = '17154'; 
+UPDATE spell dbc SET DurationIndex = '0', AuraDescription_Lang_Mask = '16712190', NameSubtext_Lang_Mask = '16712190' where ID = '18097';
```
[the remark this answers is not in the document]
Comment (Maxproz) [reply data/sql/updates/pending_db_world/rev_1620335015933017000.sql]: I was originally formatting it that way, but then I was getting compile errors. 
It turns out they are backticks and not commas.

Ill go over the sql coding standard, ty.

**Comment by FrancescoBorzi** [comment]
@azerothcore/testers @pklloveyou @mpfans @Footman @Sedatyf can anyone please check this?

**Review by Stifler82 — CHANGES_REQUESTED** [review]
Chenge the query like this, pls..

```
INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1620335015933017000');

INSERT IGNORE INTO `spell_dbc` VALUES
(17154,0,1,0,65536,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,40,3,0,0,45,45,9,0,0,0,0,0,4,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,-1,0,0,6,6,0,1,1,0,0,0,0,2,49,0,0,0,0,1,1,0,0,0,0,0,0,0,15,22,0,0,0,0,0,0,0,0,0,0,0,0,0,0,8,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,201,0,53,0,0,"The Green Tower","","","","","","","","","","","","","","","",16712190,"Rank 1","","","","","","","","","","","","","","","",16712190,"Causes $s1 Nature damage to attackers when hit and increases Nature resistance by $s2 for $d.","","","","","","","","","","","","","","","",16712190,"Increased Nature resistance and inflicts $s1 Nature damage to attackers.","","","","","","","","","","","","","","","",16712190,0,133,1500,0,7,0,0,0,0,1,0,0,1,1,1,0,0,0,0,0,0,8,0,0,0,0,0,0,0,0),
(18097,0,0,0,192,268435456,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,40,1,0,0,0,0,9,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,-1,0,0,6,0,0,1,0,0,0,0,0,-1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,42,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,17154,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,"The Green Tower","","","","","","","","","","","","","","","",16712190,"","","","","","","","","","","","","","","","",16712188,"When struck in combat has a 1% chance of raising a thorny shield that inflicts $17154s1 Nature damage to attackers when hit and increases Nature resistance by $17154s2 for $17154d.","","","","","","","","","","","","","","","",16712190,"","","","","","","","","","","","","","","","",16712188,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,0,0,0,0,0,0,32,0,0,0,0,0,0,0,0);
UPDATE `spell_dbc` SET `ProcTypeMask` = 0, `ProcChance` = 101 where `ID` = 17154; 
UPDATE `spell_dbc` SET `DurationIndex` = 0, `AuraDescription_Lang_Mask` = 16712190, `NameSubtext_Lang_Mask` = 16712190 where `ID` = 18097;
```

**Comment by Maxproz** [comment]
> So @Maxproz..
> I have tested your Fix and I have found that it does not work.
> 
> I checked the changes you made, and the reason why it doesn't work is that **although the Chance of proc you entered is correct (ie 101%)**, the problem is that **the triggers that that proc chance "works on" are missing**.
> So I took the liberty to re-work what you did, and now the Spell / Shield works.
> 
> **Video Test of my changes:** [The Green Tower](https://youtu.be/lwMrB0_sxDA)
> 
> In addition, in your code earlier, you used "quotes" (**" "**) instead you should use "apex" (**' '**)
> **(I hadn't noticed it before)**
> 
> ```
> REPLACE INTO `spell_dbc` VALUES
> (17154, 0, 1, 0, 65536, 0, 0, 33554432, 128, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1188520, 101, 0, 0, 45, 45, 9, 0, 0, 0, 0, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -1, 0, 0, 6, 6, 0, 1, 1, 0, 0, 0, 0, 2, 49, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 15, 22, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 8, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 201, 0, 53, 0, 0, 'The Green Tower','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Rank 1','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Causes $s1 Nature damage to attackers when hit and increases Nature resistance by $s2 for $d.', null, null, null, null, null, null, null, null, 0, 0, 0, 0, 0, 0, 0, 16712190, 'Increased Nature resistance and inflicts $s1 Nature damage to attackers.','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 0, 133, 1500, 0, 7, 0, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 8, 0, 0, 0, 0, 0, 0, 0, 0),
> (18097, 0, 0, 0, 192, 268435456, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1188520, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -1, 0, 0, 6, 0, 0, 1, 0, 0, 0, 0, 0, -1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 42, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 17154, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 'The Green Tower','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190,'','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'When struck in combat has a 1% chance of raising a thorny shield that inflicts $17154s1 Nature damage to attackers when hit and increases Nature resistance by $17154s2 for $17154d.', null, null, null, null, null, null, null, null, 0, 0, 0, 0, 0, 0, 0, 16712190,'','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 32, 0, 0, 0, 0, 0, 0, 0, 0);
> ```
> 
> **So the correct code is this here I did.**
> 
> You can just make COPY and PASTE

How I thought it was working was, 
Spell 18097 has 1% proc chance, with Proc OnMeleeAttackRecieved, and OnPhysicalAbilityDamageTaken.
It has spell effect SPELL_AURA_PROC_TRIGGER_SPELL and triggers spell 17154.
17154 is called and succeeds with a 101% chance, it doesn't matter what the proc was, since its already been triggered. 
It made sense to me, because why would we check the same exact proc type twice?

2 other shields also have a none procs set on the spells triggered from the equip aura 
Wall of the Dead (1979) - Aura Spell 19409 Triggers (18828 - None procs)
Skullflame Shield (1168) - Aura Spell 11815 Triggers (11817 - None procs)

**Comment by Stifler82** [comment]
@Maxproz 
it is useless for you to ask me for a review, I have already told you in my last message that I have tested this code of yours, and the shield does not work with this code.
So why should I approve something that doesn't work?

In addition, the structure / syntax of the code is wrong.
By convention **(in order to avoid DB errors)**, we use:
- `UPDATE`
- `DELETE FROM`
`INSERT INTO`
- `REPLACE`

Although, basically `REPLACE` and `INSERT IGNORE INTO`, they are the same thing, although it is not exactly so.
Because `REPLACE` overwrites the entire code, while `INSERT IGNORE INTO` also overwrites the code but tells the DB to ignore ID or ENTRY if it is a duplicate.

So on this command we could also let it pass.


But i already explained to you that we can't accept: the use of `"quotes" (" ")` instead you should use `'apex' (' ')`
And anyway, beyond the correct syntax or not.
The fundamental problem is that the values of that code do not give the final result, that is: They do not fix the shield.
because I tried your fix several time and it doesn't works..
Sorry...

On the contrary, my solution Fix the shield.
**(as the video I made also demonstrates, which shows that the shield is working properly.)**

**Review by Stifler82 — CHANGES_REQUESTED** [review]
Maybe I'm not able to explain myself well, I told you that:
(Beyond the syntax's concept), the values you entered are incorrect, the right ones, are the ones I gave you in this query!
So, if you want the PR approved, you need to enter this query.
Because with your query, shield spell **DOES NOT WORK.**
```
REPLACE INTO `spell_dbc` VALUES
(17154, 0, 1, 0, 65536, 0, 0, 33554432, 128, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1188520, 101, 0, 0, 45, 45, 9, 0, 0, 0, 0, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -1, 0, 0, 6, 6, 0, 1, 1, 0, 0, 0, 0, 2, 49, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 15, 22, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 8, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 201, 0, 53, 0, 0, 'The Green Tower','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Rank 1','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'Causes $s1 Nature damage to attackers when hit and increases Nature resistance by $s2 for $d.', null, null, null, null, null, null, null, null, 0, 0, 0, 0, 0, 0, 0, 16712190, 'Increased Nature resistance and inflicts $s1 Nature damage to attackers.','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 0, 133, 1500, 0, 7, 0, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 8, 0, 0, 0, 0, 0, 0, 0, 0),
(18097, 0, 0, 0, 192, 268435456, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1188520, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -1, 0, 0, 6, 0, 0, 1, 0, 0, 0, 0, 0, -1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 42, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 17154, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 'The Green Tower','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190,'','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 'When struck in combat has a 1% chance of raising a thorny shield that inflicts $17154s1 Nature damage to attackers when hit and increases Nature resistance by $17154s2 for $17154d.', null, null, null, null, null, null, null, null, 0, 0, 0, 0, 0, 0, 0, 16712190,'','','','','','','','','', 0, 0, 0, 0, 0, 0, 0, 16712190, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 32, 0, 0, 0, 0, 0, 0, 0, 0);
```

**Comment by Maxproz** [comment]
@Stifler82 

I'm sorry let me explain from before. 
I requested review from you because I brought up some fair points and was trying to have a conversation with you. 

However, in your subsequent responses it became clear that you were incapable of having an intelligent conversation so I gave up.

For me, I want to commit the best fix possible, not commit the first thing that ends up working. What are people going to say in the future when they are trying to fix something similar and see inconsistencies between shields? In theory, there is a good chance that both of our fixes work. Being an engineer is not always about 'just copy and paste' like you keep saying. 

Frankly, if you are the first person that people run into on their first commit this project will have a hard time keeping new people that decide to try contributing. Please stop commenting on my reviews, I don't learn anything.

Out of respect for shin and hacki I have held back as much as possible in this response.

**Comment by Stifler82** [comment]
> @Stifler82
> 
> I'm sorry let me explain from before.
> I requested review from you because I brought up some fair points and was trying to have a conversation with you.
> 
> However, in your subsequent responses it became clear that you were incapable of having an intelligent conversation so I gave up.
> 
> For me, I want to commit the best fix possible, not commit the first thing that ends up working. What are people going to say in the future when they are trying to fix something similar and see inconsistencies between shields? In theory, there is a good chance that both of our fixes work. Being an engineer is not always about 'just copy and paste' like you keep saying.
> 
> Frankly, if you are the first person that people run into on their first commit this project will have a hard time keeping new people that decide to try contributing. Please stop commenting on my reviews, I don't learn anything.
> 
> Out of respect for shin and hacki I have held back as much as possible in this response.

@Maxproz 
I can't explain the reason for your aggressive attitude, I didn't tell you anything bad, just yesterday I tried your code, and the shield didn't work.

In any case, I find myself forced to apologize to you, just because I am scrupulous, I just tried your code again, and now the shield works instead.
Honestly, I have no idea why it didn't work yesterday and now it does.
Probably a client cache error. **(it's the only thing I can attribute the error to)**.

So, since you fixed the Syntax, and since after a further test I realized that the Fix works, so for me the PR is fine.

In any case, **(since you said you want to learn).**
Generally speaking **(to avoid doing heavy codes) and therefore avoid doing Wall of the Text Codes.**
it is preferable **(but not mandatory)** to avoid overcoding, 

Like in your code, in your code, in a first Step you are making an `INSERT IGNORE INTO`, then you making `2 UPDATES` on that `INSERT IGNORE INTO` this is a sort of **"Double Work"** on the same IDs that you are making on the DB. **just FYI.**

**Review by Stifler82 — APPROVED** [review]

**Comment by Hacki95** [comment]
@azerothcore/testers Can somebody give this a quick test?

**Comment by Yehonal** [comment]
@Stifler82 we cannot use REPLACE INTO because in that way we don't have the evidence of what is changed. We MUST use the UPDATE queries right after the insert ignore

please follow the guidelines here: https://www.azerothcore.org/wiki/how-to-import-dbc-data-in-db

**Comment by Yehonal** [comment]
@Maxproz thanks for the contribution!

could you or someone else provide video evidence of the test with a before/after?

**Comment by FrancescoBorzi** [comment]
about our SQL standards, I've seen different opinions here. This is what we do:

- use `UPDATE` whenever possible
- always use a `DELETE` before an `INSERT` (with one only exception, see below)
- the only exception is `spell_dbc` where we use `INSERT IGNORE` **and** `UPDATE` together, there is a specific reason why in this table we act differently
- NEVER use `REPLACE`


For more details please read: https://www.azerothcore.org/wiki/sql-standards

File: data/sql/updates/pending_db_world/rev_1620335015933017000.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+UPDATE `spell_dbc` SET `ProcTypeMask` = 0, `ProcChance` = 101 WHERE `ID` = 17154; 
+UPDATE `spell_dbc` SET `DurationIndex` = 0, `AuraDescription_Lang_Mask` = 16712190, `NameSubtext_Lang_Mask` = 16712190 WHERE `ID` = 18097;
```
[the remark this answers is not in the document]
Comment (Maxproz) [reply data/sql/updates/pending_db_world/rev_1620335015933017000.sql:8]: @Yehonal 

The original proc chance calculation was probably working it was probably around 3/10000 chance of happening? 
0.01 * 0.03 == 0.0003  == 3/10000

ProcTypeMask = 0, is not needed, it was just added to 17154 for consistency between other shields that have similar proc sequences, and also for clarification, it doesn't appear to be used here at all. The ProcTypeMask is set in 18097.

For duration index = 0 on 18097 it is set and handled inside 17154 so its not needed there.

For AuraDescription_Lang_Mask and NameSubtext_Lang_Mask, these we're set automatically when I exported from wow spell editor. They don't appear to do anything at all. But 16712190 was set on other options in the columns so I had them all match.

[the remark this answers is not in the document]
Comment (Maxproz) [reply data/sql/updates/pending_db_world/rev_1620335015933017000.sql:8]: Its save to assume the Lang_Masks are related to the Spell Name: Spell Rank: Spell ToolTip: Spell Description etc. 

But localization doesn't appear to be handeled here at all. So it probably doesn't matter if its there or not.

[the remark this answers is not in the document]
Comment (Maxproz) [reply data/sql/updates/pending_db_world/rev_1620335015933017000.sql:8]: Most likely that is handeled by  client im guessing

Comment (Yehonal) [inline data/sql/updates/pending_db_world/rev_1620335015933017000.sql:8]:
```suggestion
UPDATE `spell_dbc` SET `DurationIndex` = 0 WHERE `ID` = 18097;
```

**Review by Yehonal — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1620335015933017000.sql (hunk 2 above)
Comment (Maxproz) [reply data/sql/updates/pending_db_world/rev_1620335015933017000.sql]: If you ever find out exactly what those enums apply, let me know. I am curious about why they were changed on my extraction.

**Comment by FrancescoBorzi** [comment]
what's the status of this?

**Comment by Maxproz** [comment]
> what's the status of this?

core-build / ubuntu-20.04-clang9-without-modules-extra_logs-false (pull_request) Failing after 50
I assume that there is a problem with this PC and my code should work?

I guess we are just waiting for someone to test.

**Comment by FrancescoBorzi** [comment]
> > what's the status of this?
> 
> core-build / ubuntu-20.04-clang9-without-modules-extra_logs-false (pull_request) Failing after 50
> I assume that there is a problem with this PC and my code should work?
> 
> I guess we are just waiting for someone to test.

I can confirm, the failure with clang-9 is not related with your PR

**Comment by Sedatyf** [comment]
I was in a 10 minutes long combat with a boar and nothing happened. I know 1% is not that much, but I can say that it's not working (not 100% confident on this).
Here steps I did: 
- Applied SQL request in my database
- Launched worldserver
- Did steps provided by @Maxproz 
- Nothing happened 
Feel free to reach me if you have any questions

**Comment by Maxproz** [comment]
One thing I forgot to mention. 
The proc is set to only go off for "1% chance on take melee physical damage"

As a result if a lvl 1 boar is attacking you from the front, and you block, parry, dodge, or it misses. 
The proc chance doesn't even calculate for that swing. 

Up to you what you want me to do @FrancescoBorzi

**Comment by Sedatyf** [comment]
Well boar hit me a lot of time during this 10 minutes fight. Indeed I blocked and it missed me, but I took a lot of damage but nothing happened.

**Comment by Maxproz** [comment]
> Well boar hit me a lot of time during this 10 minutes fight. Indeed I blocked and it missed me, but I took a lot of damage but nothing happened.

Yeah I know I was watching stream, all good. 
For future reference, you can avoid block and parry by facing away from the enemy.

**Comment by Sedatyf** [comment]
> Yeah I know I was watching stream, all good.
> For future reference, you can avoid block and parry by facing away from the enemy.

You should have say something when I was testing it 😄 
And indeed it's a great idea, I didn't thought about it.

**Comment by Sedatyf** [comment]
@Maxproz can you have a look on this?

**Comment by Maxproz** [comment]
> @Maxproz can you have a look on this?

Ill take a look monday morning

**Comment by Yehonal** [comment]
Any news here?

**Comment by Sedatyf** [comment]
I was waiting to @Maxproz but I didn't have any news since

**Comment by Nefertumm** [comment]
Any news on this?

**Comment by FrancescoBorzi** [comment]
@Ragelezz what's the status of this PR?

**Comment by Ragelezz** [comment]
Same here, this can be closed since I created another PR, this one is not created by me so I don't have acces to edit this.

**Pull request closed without merging**